text/0000-add-lifecycle-config.md

```diff
@@ -0,0 +1,117 @@
+
+- What use cases does it support?
+  - Platforms that run all phases in a single kubernetes pod will have the ability to write configuration before and after each phase for the upcoming phases using a volume.
+
+- What is the expected outcome?
+  - Platforms that use the buildpacks.io published images like `buildpacksio/lifecycle` will have a method to configure how lifecycle behaves by writing a file in a configurable location.
+  - This will open up platform operators to be able to generate a configuration file based on information that may not be available at pod creation time but is available once the source code is available.
+  - Some platforms may move from setting `CNB_*` env vars to producing a `config.toml`.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Target persona: platform operator, platform implementor
+
+A new `config.toml` will be read by lifecycle. By default, this path searched will be `/cnb/config.toml`. `CNB_CONFIG_PATH` would be available to override this location.
+
+An example `config.toml` might be produced by the platform with various inputs (target source code, platform specific configuration, etc.):
+```
+[config]
+order-path = "/artifacts/generated-order.toml"
+no-color = true
+log-level = "debug"
+```
+
+An example config resolution would be, in prioritized order, CLI flag, os env `CNB_ORDER_PATH`, `[config.toml][config]order_path`, then the `lifecycle` default of `/cnb/order.toml`.
+
+The keys of the `[config]` table are expected to match the env var names, minus the `CNB_` prefix and kebab-case.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Lifecycle will read in `CNB_CONFIG_PATH` or the default of `/cnb/config.toml` and prioritize the values found over lifecycle's defaults and before present env vars.
+
+If there is no configuration found, lifecycle will resolve configuration as it does today. That order is cli flag, env var, default.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+Another configuration file.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- What other designs have been considered?
+  - Lifecycle could introduce a `prepare` phase that processes a project's `project.toml` to generate the input files needed by detector, much like `pack` does today.
+  - Lifecycle could load env vars from a known file location. e.g. `source lifecycle.env` or similar prior to resolving lifecycle configuration.
+  - Lifecycle could only allow for non-path based inputs like `log-level` in `config.toml`, but add an additional path to to a known platform location. Example would be `/cnb/config/order.toml` for `CNB_ORDER_PATH`. This would allow platforms to use a volume at `/cnb/config` and write files that are fed into lifecycle in addition to a `config.toml` for non-path based configuration.
```

Comment:
```suggestion
  - Lifecycle could only allow for non-path based inputs like `log-level` in `config.toml`, but add an additional path to a known platform location. Example would be `/cnb/config/order.toml` for `CNB_ORDER_PATH`. This would allow platforms to use a volume at `/cnb/config` and write files that are fed into lifecycle in addition to a `config.toml` for non-path based configuration.
```